zone/aggro.cpp

```diff
@@ -564,6 +564,55 @@ bool Mob::CheckWillAggro(Mob *mob) {
 	return false;
 }
 
+int EntityList::FleeAllyCount(Mob *attacker, Mob *exclude)
+{
+	// Return a list of how many NPCs of the same faction or race are within aggro range of the given exclude Mob.
+	if (!attacker) {
+		return 0;
+	}
+
+	int count = 0;
+
+	for (auto it = npc_list.begin(); it != npc_list.end(); ++it) {
+		NPC *mob = it->second;
+		if (!mob || (mob == exclude)) {
+			continue;
+		}
+
+		float AggroRange = mob->GetAggroRange();
+		float AssistRange = mob->GetAssistRange();
+
+		if(AssistRange > AggroRange) {
+			AggroRange = AssistRange;
+		}
+
+		// Square it because we will be using DistNoRoot
+		AggroRange *= AggroRange;
+
+		if (DistanceSquared(mob->GetPosition(), exclude->GetPosition()) > AggroRange) {
+			continue;
+		}
+
+		// If exclude doesn't have a faction, check for buddies based on race. Also exclude common factions such as noob monsters, indifferent, kos, kos animal
+		if(exclude->GetPrimaryFaction() != 0 &&
+			exclude->GetPrimaryFaction() != 5023 && exclude->GetPrimaryFaction() != 5032 && exclude->GetPrimaryFaction() != 5013 && exclude->GetPrimaryFaction() != 5014) {
```

Comment:
Should we be _really_ be hard coding these?

Reply:
Nope, but I am not sure how else to reference these specially as different servers could have different ids.

Reply:
Make them rules? e.g.
RuleI(Aggro, IndifferentFactionID)?

Reply:
We can use a string-based rule so we can have it pipe-separated like `1|2|3` or something it they want multiple factions.